eth_account_snapshot.go

```diff
@@ -89,11 +89,11 @@ func (as *EthAccountSnapshot) Resolve(p []string) (interface{}, []string, error)
 	case "balance":
 		return as.Balance, nil, nil
 	case "codeHash":
-		return fmt.Sprintf("0x%x", as.CodeHash), nil, nil
+		return &node.Link{Cid: keccak256ToCid(MEthCodeHash, as.CodeHash)}, nil, nil
 	case "nonce":
 		return as.Nonce, nil, nil
 	case "root":
-		return fmt.Sprintf("0x%x", as.Root), nil, nil
+		return &node.Link{Cid: keccak256ToCid(MEthStorageTrie, as.Root)}, nil, nil
```

Comment:
@hermanjunge not sure if it's correct. Seems like the storage root is not actually keccak encoded, only the keys are. I think this needs reverting to `rawdataToCid`?

Reply:
It is encoded. Each key is `keccak256(val)`

Take for example this very test file you are adding `eth-storage-trie-rlp-000dd0`, its contents, if you `xxd -p -c1000` it, are

```
f85180808080a07a6b50fdf61a20ef57cfd22a051c3bfc21185852b7f6a86dac72ac44c2b2291e80808080808080a02db9634453975816c20676cdaca000d9da7db78f947eeaaaea53db1a3e18f94380808080
```

Which encoded to bytes, and hashed correspond to

```
000dd0b4777a15d7dd300ff5c8d3eb19ec0038c8f00958494863c7f071847161
```

Reply:
Maybe your concern is on the merkle path. Which is correct. In the state trie, you do `keccak256(address)` to reach a leave, whereas in the storage trie, you don´t.